src/app/components/LibraryNavigator/index.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { createStackNavigator } from '@react-navigation/stack';
+import Routes from '@constants/routes';
+import BookList from '@app/screens/BookList';
+import BookDetail from '@app/screens/BookDetail';
+import { RootStackParamList } from '@interfaces/navigation';
+import { stackNavigatorConfig } from '@config/navigation';
+
+function LibraryNavigator() {
+  const { Navigator, Screen } = createStackNavigator<RootStackParamList>();
```

Comment:
We don't really use reactnavigaton5 in any projects, so we don't have a convention on where to put these. But I'd move it to the same place where you have the tabnavigator.

```
const LibraryStack = createStackNavigator();
const MainTabs = createBottomTabNavigator();

function LibraryStackScreen() {
  return (
    <LibraryStack.Navigator    >
      <LibraryStack.Screen   />
      <LibraryStack.Screen     />
    </LibraryStack.Navigator>
  );
}

function TabNavigatorScreen() {
  return (
    <MainTabs.Navigator>
      <MainTabs.Screen   component={LibraryStackScreen}/>
      <MainTabs.Screen     component={Wishlist}      />
    </MainTabs.Navigator>
  );
}
```

Reply:
I agree with Marian. Create a  new component `AppNavigator` in `@components/AppNavigator` to handle all  the  stack, tabs, etc... logic.
And then, in the `@app/index.tsx`:
```
const App = () => {
  return (
    <NavigationContainer>
      <StatusBar barStyle="light-content" />
      <AppNavigator />
    </NavigationContainer>
  );
};
```
`AppNavigator` ===  `TabNavigatorScreen `

Reply:
![image](https://user-images.githubusercontent.com/22043902/83291452-f6305c00-a1ad-11ea-9542-3c8620e534c0.png)